recipes/simple-websocket-server/all/conanfile.py

```diff
@@ -27,10 +27,17 @@ def _source_subfolder(self):
         return "source_subfolder"
 
     def requirements(self):
+        # only version 2.0.2 upwards is able to build against asio 1.18.0 or higher
         if self.options.use_asio_standalone:
-            self.requires("asio/1.16.1")
+            if self.version <= "2.0.1":
+                self.requires("asio/1.16.1")
+            else:
+                self.requires("asio/1.18.0")
```

Comment:
I'm wondering, if some other recipe or the profile requires `asio/1.16.1`, then the build of versions <= 2.0.1 will fail.
Should the recipe have a check in `build`?
something like:
```python
def build(self):
    if tools.Version(self.version) <= "2.0.1":
        if tools.Version(self.deps_cpp_info["asio"] >= "1.18.0":
            raise ConanInvalidConfiguration("simple-websocket-server versions <=2.0.1 require asio < 1.18.0")
    else:
        if tools.Version(self.deps_cpp_info["asio"] < "1.18.0":
            raise ConanInvalidConfiguration("simple-websocket-server versions >2.0.1 require asio >= 1.18.0")
```

I'm doing something similar in my cpython recipe for mpdecimal.

Reply:
Added a check.